scan/src/components/tx-index/IndexDataMsg.re

```diff
@@ -0,0 +1,238 @@
+         | Some(calldataKVs) =>
+           <KVTable
+             rows={
+               calldataKVs
+               ->Belt_Array.map(({fieldName, fieldValue}) =>
+                   [KVTable.Value(fieldName), KVTable.Value(fieldValue)]
+                 )
+               ->Belt_List.fromArray
+             }
+           />
+         | None =>
+           <Text
+             value="Could not decode calldata."
+             spacing={Text.Em(0.02)}
+             nowrap=true
+             ellipsis=true
+             code=true
+             block=true
+           />
+         }}
+      </Col.Grid>
+      <Col.Grid col=Col.Six mbSm=24>
+        <Heading value="Request Validator Count" size=Heading.H5 marginBottom=8 />
+        <Text value={request.askCount |> string_of_int} />
+      </Col.Grid>
+      <Col.Grid col=Col.Six>
+        <Heading value="Sufficient Validator Count" size=Heading.H5 marginBottom=8 />
+        <Text value={request.minCount |> string_of_int} />
+      </Col.Grid>
+    </Row.Grid>;
+  };
+};
+
+module RequestFailMsg = {
+  [@react.component]
+  let make = (~request: TxSub.Msg.Request.fail_t) => {
+    <Row.Grid>
+      <Col.Grid col=Col.Six mb=24>
+        <Heading value="Owner" size=Heading.H5 marginBottom=8 />
+        <AddressRender address={request.sender} />
+      </Col.Grid>
+      <Col.Grid col=Col.Six mb=24>
+        <Heading value="Oracle Script" size=Heading.H5 marginBottom=8 />
+        <div className={CssHelper.flexBox()}>
+          <TypeID.OracleScript id={request.oracleScriptID} />
+        </div>
+      </Col.Grid>
+      <Col.Grid mb=24>
+        <Heading value="Calldata" size=Heading.H5 marginBottom=8 />
+        <div className={CssHelper.flexBox()}>
+          <Text value={request.calldata |> JsBuffer.toHex} color=Colors.gray7 />
+          <HSpacing size=Spacing.sm />
+          <CopyRender width=14 message={request.calldata |> JsBuffer.toHex} />
+        </div>
+      </Col.Grid>
+      <Col.Grid col=Col.Six mbSm=24>
+        <Heading value="Request Validator Count" size=Heading.H5 marginBottom=8 />
+        <Text value={request.askCount |> string_of_int} />
+      </Col.Grid>
+      <Col.Grid col=Col.Six>
+        <Heading value="Sufficient Validator Count" size=Heading.H5 marginBottom=8 />
+        <Text value={request.minCount |> string_of_int} />
+      </Col.Grid>
+    </Row.Grid>;
+  };
+};
+
+module ReportMsg = {
+  [@react.component]
+  let make = (~report: TxSub.Msg.Report.t) => {
+    <Row.Grid>
+      <Col.Grid col=Col.Six mb=24>
+        <Heading value="Owner" size=Heading.H5 marginBottom=8 />
+        <AddressRender address={report.reporter} />
+      </Col.Grid>
+      <Col.Grid col=Col.Six mb=24>
+        <Heading value="Request ID" size=Heading.H5 marginBottom=8 />
+        <div className={CssHelper.flexBox()}> <TypeID.Request id={report.requestID} /> </div>
+      </Col.Grid>
+      <Col.Grid>
+        <Heading value="Raw Data Report" size=Heading.H5 marginBottom=8 />
+        <KVTable
```

Comment:
Can we add request id as well?

Reply:
There still has the requestID on line 218, but I've removed the container.